### PR TITLE
enhance security , severity:high, (eval input,prevent shell rce)

### PR DESCRIPTION
**Pull request by gellanyhassan0**
am fix at first two security high severity bug in project  

1- (INPUT not safe) 
 the input function not evaluate and this lead to Preventing input vulnerabilities https://www.geeksforgeeks.org/vulnerability-input-function-python-2-x/
mitigation :
should be eval by type of  input function 
https://cwe.mitre.org/data/definitions/20.html
https://cwe.mitre.org/data/definitions/1287.html
https://cwe.mitre.org/data/definitions/79.html

----------------------------------------------------------------------------------------

2- (shell of PREPROCESS not safe)
shell_equals_true lead to Remote code execution (RCE)  and shell_injection attacks
mitigation : 
disable it to false
https://security.openstack.org/guidelines/dg_use-subprocess-securely.html
https://security.openstack.org/guidelines/dg_avoid-shell-true.html
https://cwe.mitre.org/data/definitions/78.html

### CODE REVIEW

File: ci/build/build-docker-images.py [hunk 1]
```diff
@@ -111,11 +111,11 @@ def _get_commit_sha():
 def _configure_human_version():
     global _get_branch
     global _get_commit_sha
-    fake_branch_name = input(
+    fake_branch_name = str(input(
```
Comment (jjyao) [inline ci/build/build-docker-images.py:114]: If I understand correctly, input() doesn't have security issue in Python 3? Ray doesn't support Python 2.X so it should be fine.

**Review by jjyao — COMMENTED** [review]
Thanks for the contribution.

For the shell related changes, seems some tests are failing. I guess some code relies on `shell=True` now.

File: ci/build/build-docker-images.py (hunk 1 above)
Comment (gellanyhassan0) [reply ci/build/build-docker-images.py:114]: @jjyao input should be validation by type or by function input is safe

Comment (jjyao) [reply ci/build/build-docker-images.py:114]: Hmm, I'm not sure if I understand what you said. Do we still need to add `str(input())` if we are using Python 3?

Comment (gellanyhassan0) [reply ci/build/build-docker-images.py:114]: @jjyao this up to you , you review this https://cisserv1.towson.edu/~cssecinj/modules/cs0/cs0-input-validation-python/

**Pull request closed without merging**